impeller/toolkit/egl/context.cc

```diff
@@ -0,0 +1,59 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "impeller/toolkit/egl/context.h"
+
+#include "impeller/toolkit/egl/surface.h"
+
+namespace impeller {
+namespace egl {
+
+Context::Context(EGLDisplay display, EGLContext context)
+    : display_(display), context_(context) {}
+
+Context::~Context() {
+  if (context_ != EGL_NO_CONTEXT) {
+    if (::eglDestroyContext(display_, context_) != EGL_TRUE) {
+      IMPELLER_LOG_EGL_ERROR;
+    }
+  }
+}
+
+bool Context::IsValid() const {
+  return context_ != EGL_NO_CONTEXT;
+}
+
+const EGLContext& Context::GetHandle() const {
+  return context_;
+}
+
+bool Context::MakeCurrent(const Surface& surface) const {
+  if (context_ == EGL_NO_CONTEXT) {
+    return false;
+  }
+  const auto result = ::eglMakeCurrent(display_,             //
```

Comment:
@iskakaushik recently found that this call can be expensive/blocking. If the idea is that it'll already be guarded by other changes he's made in the Android embedding that's fine, but otherwise we should guard it here as well so that if the current context is already thecontext we want we avoid trying to make it current again.

Reply:
I believe the cause for why that was expensive was that the call flushed previous commands and waited for them to be committed. There is nothing in the toolkit that attempts to sidestep this but perhaps such a fix should be applied here as well. I'll investigate if it makes sense. Applying it here will benefit all EGL platforms.

Reply:
Done.